ctapipe_io_magic/__init__.py

```diff
@@ -351,54 +384,84 @@ def parse_run_info(self):
             'MRawRunHeader.fTelescopeNumber',
         ]
 
-        run_info = self.file_['RunHeaders'].arrays(
-            runinfo_array_list, library="np")
-        run_number = int(run_info['MRawRunHeader.fRunNumber'][0])
-        run_type = int(run_info['MRawRunHeader.fRunType'][0])
-        telescope_number = int(run_info['MRawRunHeader.fTelescopeNumber'][0])
-
-        # a note about run numbers:
-        # mono data has run numbers starting with 1 or 2 (telescope dependent)
-        # stereo data has run numbers starting with 5
-        # if both telescopes are taking data with no L3,
-        # also in this case run number starts with 5 (e.g. muon runs)
-
-        # Here the data types (from MRawRunHeader.h)
-        # std data = 0
-        # pedestal = 1 (_P_)
-        # calibration = 2 (_C_)
-        # domino calibration = 3 (_L_)
-        # linearity calibration = 4 (_N_)
-        # point run = 7
-        # monteCarlo = 256
-        # none = 65535
-
-        mc_data_type = 256
-
-        if run_type == mc_data_type:
-            is_mc = True
-        else:
-            is_mc = False
+        run_numbers = []
+        is_simulation = []
+        telescope_numbers = []
+        datalevels = []
+
+        for rootf in self.files_:
+
+            run_info = rootf['RunHeaders'].arrays(
+                runinfo_array_list, library="np")
+            run_number = int(run_info['MRawRunHeader.fRunNumber'][0])
+            run_type = int(run_info['MRawRunHeader.fRunType'][0])
+            telescope_number = int(run_info['MRawRunHeader.fTelescopeNumber'][0])
+
+            # a note about run numbers:
+            # mono data has run numbers starting with 1 or 2 (telescope dependent)
+            # stereo data has run numbers starting with 5
+            # if both telescopes are taking data with no L3,
+            # also in this case run number starts with 5 (e.g. muon runs)
+
+            # Here the data types (from MRawRunHeader.h)
+            # std data = 0
+            # pedestal = 1 (_P_)
+            # calibration = 2 (_C_)
+            # domino calibration = 3 (_L_)
+            # linearity calibration = 4 (_N_)
+            # point run = 7
+            # monteCarlo = 256
+            # none = 65535
+
+            mc_data_type = 256
```

Comment:
This is a constant, probably better defined somewhere at the toplevel.

Reply:
Done.